File: java/leetcode_grind/Day566.java
package leetcode_grind;

public class Day566 {
    // https://leetcode.com/problems/append-characters-to-string-to-make-subsequence/description
    static class Solution {
        public int appendCharacters(String s, String t) {
            int first = 0, longestPrefix = 0;
            while (first < s.length() && longestPrefix < t.length()) {
                if (s.charAt(first) == t.charAt(longestPrefix)) {
                    longestPrefix++;
                }
                first++;
            }
            return t.length() - longestPrefix;
        }
    }
}

File: python/leetcode_grind/day_566.py
# https://leetcode.com/problems/is-subsequence/description/
class Solution1:
    def isSubsequence(self, s: str, t: str) -> bool:
        if not len(s):
            return True

        i, j = 0, 0
        while i < len(s) and j < len(t):
            if s[i] == t[j]:
                i+=1
            j+=1
        return i == len(s)